sim/warlock/curses.go

```diff
@@ -212,7 +217,9 @@ func (warlock *Warlock) registerBaneOfDoomSpell() {
 			},
 			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
 				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeTick)
-				//TODO: 20% Chance to summon Ebon Imp
+				if sim.Proc(0.2+ebonImpBonusSummon, "Ebon Imp") {
+					warlock.EbonImp.EnableWithTimeout(sim, warlock.EbonImp, time.Second*15)
+				}
```

Comment:
This is something I thought about as well. In looking online and testing the imp lasts exactly 15 seconds, and since bane of doom is only one per warlock the only way you could get multiple imps is if bane of doom benefitted from haste and the ticks were reduced enough where the imps overlapped. I don't think the spell benefits from haste though.